pkgs/by-name/um/umockdev/package.nix

```diff
@@ -85,7 +94,7 @@ stdenv.mkDerivation (finalAttrs: {
     "-Dgtk_doc=true"
   ];
 
-  doCheck = true;
+  doCheck = withSystemd;
```

Comment:
Related to the above comments, this should be based on `udev.withSystemd` rather than a package argument. And we need a `nativeCheckInputs` to add `systemdMinimal` to the inputs for the check phase so that `udevadm` can be found. This of course does not need anything like `lib.optionals` since check inputs are automatically discarded when `doCheck = false`.